music_assistant/providers/audiobookshelf/__init__.py

```diff
@@ -0,0 +1,534 @@
+            mass_audiobook.metadata.languages = UniqueList([abs_audiobook.media.metadata.language])
+        mass_audiobook.metadata.release_date = abs_audiobook.media.metadata.published_date
+        if abs_audiobook.media.metadata.genres is not None:
+            mass_audiobook.metadata.genres = set(abs_audiobook.media.metadata.genres)
+
+        # chapters
+        chapters = []
+        for idx, chapter in enumerate(abs_audiobook.media.chapters):
+            chapters.append(
+                MediaItemChapter(
+                    position=idx + 1,  # chapter starting at 1
+                    name=chapter.title,
+                    start=chapter.start,
+                    end=chapter.end,
+                )
+            )
+        mass_audiobook.metadata.chapters = chapters
+
+        mass_audiobook.metadata.explicit = abs_audiobook.media.metadata.explicit
+        progress, finished = await self._client.get_audiobook_progress_ms(abs_audiobook.id_)
+        if progress is not None:
+            mass_audiobook.resume_position_ms = progress
+            mass_audiobook.fully_played = finished
+
+        # cover
+        base_url = f"{self.config.get_value(CONF_URL)}"
+        api_url = f"/api/items/{abs_audiobook.id_}/cover?token={self._client.token}"
+        cover_url = f"{base_url}{api_url}"
+        mass_audiobook.metadata.images = UniqueList(
+            [MediaItemImage(type=ImageType.THUMB, path=cover_url, provider=self.lookup_key)]
+        )
+
+        return mass_audiobook
+
+    async def get_library_audiobooks(self) -> AsyncGenerator[Audiobook, None]:
+        """Get Audiobook libraries."""
+        async for abs_audiobook in self._client.get_all_audiobooks():
+            mass_audiobook = await self._parse_audiobook(abs_audiobook)
+            yield mass_audiobook
+
+    async def get_audiobook(self, prov_audiobook_id: str) -> Audiobook:
+        """Get a single audiobook."""
+        abs_audiobook = await self._client.get_audiobook(prov_audiobook_id)
+        return await self._parse_audiobook(abs_audiobook)
+
+    async def get_stream_details(
+        self, item_id: str, media_type: MediaType = MediaType.TRACK
+    ) -> StreamDetails:
+        """Get stream of item."""
+        if media_type == MediaType.PODCAST_EPISODE:
+            return await self._get_stream_details_podcast_episode(item_id)
+        elif media_type == MediaType.AUDIOBOOK:
+            return await self._get_stream_details_audiobook(item_id)
+        raise MediaNotFoundError("Stream unknown")
+
+    async def _get_stream_details_audiobook(self, audiobook_id: str) -> StreamDetails:
+        """Only single audio file in audiobook."""
+        abs_audiobook = await self._client.get_audiobook(audiobook_id)
+        tracks = abs_audiobook.media.tracks
+        if len(tracks) == 0:
+            raise MediaNotFoundError("Stream not found")
+        if len(tracks) > 1:
+            logging.warning("MASS probably only supports single file base audiobooks")
```

Comment:
```suggestion
            logging.warning("Music Assistant only supports single file base audiobooks")
```

For now, we only support single file audiobooks. Let me know if this is very common or many people run into this, then I can add support to the core.